### PR TITLE
Migrate to .Net 6.0 template style

### PR DESCRIPTION
**Pull request by Blackclaws**
This switches to the .Net 6.0 template style.

There is an existing issue that the:

ReturnsSeedProjectGivenId1 Test fails but only when its being run in parallel. If its being run by itself this doesn't happen. I'm not sure why that is the case

Closes: https://github.com/ardalis/CleanArchitecture/issues/260

### CODE REVIEW

**Comment by Blackclaws** [comment]
After some checks it appears that the test data gets added, however the ids are not stable for the projects, it seems the database gets cleared but not reset between checks.

**Comment by Blackclaws** [comment]
This now explicitly calls ConfigureWebHost during CreateHost fixing the issue.

**Comment by Blackclaws** [comment]
Fixed most nullability warnings as well. There remains one from the NoOpMediator implementation.

**Comment by ardalis** [comment]
I'm also in the process of making similar updates to eShopOnWeb. I assume your fix here will fix any parallel test issues there as well? Thanks!

**Comment by Blackclaws** [comment]
Yeah I saw that. Its a bit weird and I'm not entirely sure what's happening here. I thought that calling ConfigureWebHost explicitly would solve the problem by itself (Which it seemed to do) however it came back once I fixed the nullable warnings (for whatever reason).

Only explicitly resetting the ID of the test data seemed so solve the problem. Apparently changes got persisted to the test data during the other test runs.

Intermittent test failures like these are the worst to debug.

File: src/Clean.Architecture.Web/Api/ProjectsController.cs [hunk 1]
```diff
@@ -66,10 +67,10 @@ public async Task<IActionResult> Post([FromBody] CreateProjectDTO request)
         var createdProject = await _repository.AddAsync(newProject);
 
         var result = new ProjectDTO
-        {
-            Id = createdProject.Id,
-            Name = createdProject.Name
-        };
+        (
+            id: createdProject.Id,
+            name: createdProject.Name
```
Comment (ardalis) [inline src/Clean.Architecture.Web/Api/ProjectsController.cs:72]: Why this change?

Comment (Blackclaws) [reply src/Clean.Architecture.Web/Api/ProjectsController.cs:72]: To satisfy the nullable warnings. If the fields aren't declared as nullable there is a warning if they aren't being set in the constructor. The alternative would have been to just set the fields nullable, however I didn't really see a reason to do that.

Comment (ardalis) [reply src/Clean.Architecture.Web/Api/ProjectsController.cs:72]: I'm assuming to ensure non-null by going through the constructor.

**Review by ardalis — APPROVED** [review]

**Comment by Blackclaws** [comment]
I've just noticed that it is indeed seeding the database twice which results in the test failures. I'll go back to the drawing board on that one and fix that correctly.

**Comment by ardalis** [comment]
Ah, thanks. I wish getting these apps to run with appropriate seed data, isolated from one another, weren't so difficult...

**Comment by Blackclaws** [comment]
Yeah. I'm currently working on fixing that, so expect another pull request soonish. The current solution was a kludge because I misunderstood what was happening in the first place.
The issue is that the seed code in Program.js runs as well for the DeferredHostBuilder. So you get the default seeding of the database as well as the seeding from the test database.

**Comment by Blackclaws** [comment]
Ok things are really weird when working with DeferredHostBuilder. Apparently the code in Program.js is run in parallel with the code in CreateHost for some reason. I don't really get it. This probably means stepping through one by one.

**Comment by Blackclaws** [comment]
Fixed in: https://github.com/ardalis/CleanArchitecture/pull/277

**Comment by brokenthorn** [comment]
Does [Jason Taylor's CleanArchitecture Template](https://github.com/jasontaylordev/CleanArchitecture) help?

**Comment by ardalis** [comment]
It's worth looking at. Last time I checked, it exposed the data layer too much for my taste.
